services/app-api/src/postgres/contractAndRates/updateDraftContractWithRates.ts

```diff
@@ -215,16 +234,44 @@ async function updateDraftContractWithRates(
                                         rateCertificationName:
                                             rateRevision.rateCertificationName,
                                         rateDocuments: {
-                                            create: rateRevision.rateDocuments,
+                                            create:
+                                                rateRevision.rateDocuments &&
+                                                rateRevision.rateDocuments.map(
+                                                    (d, idx) => ({
+                                                        position: idx,
+                                                        ...d,
+                                                    })
+                                                ),
                                         },
                                         supportingDocuments: {
-                                            create: rateRevision.supportingDocuments,
+                                            create:
+                                                rateRevision.supportingDocuments &&
+                                                rateRevision.supportingDocuments.map(
+                                                    (d, idx) => ({
+                                                        position: idx,
+                                                        ...d,
+                                                    })
+                                                ),
                                         },
                                         certifyingActuaryContacts: {
-                                            create: rateRevision.certifyingActuaryContacts,
+                                            create:
+                                                rateRevision.certifyingActuaryContacts &&
+                                                rateRevision.certifyingActuaryContacts.map(
+                                                    (c, idx) => ({
+                                                        position: idx,
+                                                        ...c,
+                                                    })
+                                                ),
                                         },
                                         addtlActuaryContacts: {
-                                            create: rateRevision.addtlActuaryContacts,
+                                            create:
+                                                rateRevision.addtlActuaryContacts &&
+                                                rateRevision.addtlActuaryContacts.map(
+                                                    (c, idx) => ({
+                                                        position: idx,
```

Comment:
@macrael I think we should add this handling to `insertContract` and `insertRate` also -  feels like that could be a place that introduces inconsistency. Otherwise looks good!